examples/jsm/nodes/core/NodeLib.d.ts

```diff
@@ -0,0 +1,23 @@
+import { Node } from './Node';
+
+export interface NodeLibKeyword {
+  callback: () => void;
+  cache?: object;
+}
+
+export namespace NodeLib {
+
+  export const nodes: object;
+  export const keywords: object;
+
+  export function add(node: Node): void;
+  export function addKeyword(name: string, callback: () => void, cache?: object): void;
+  export function remove(node: Node): void;
+  export function removeKeyword(name: string): void;
+  export function get(name: string): Node;
+  export function getKeyword(name: string, material: object): any;
```

Comment:
@sunag I'm unsure about `getKeyword()`. This method is called in `FunctionNode.generate()` but with `NodeBuilder` as the second parameter. However, in the `NodeLib` source file the parameter is named `material`. 

In any event, `material` is assigned to the callback function defined in `addKeyword()`. What is the intended parameter of that callback function?

Reply:
Hi @Mugen87 `getKeyword(name: string, material: object)` is outdated this should be `getKeyword(name: string, builder: object)`. 

`builder` argument can be useful to the callback return a node appropriate to a determined material and processes like vertex or fragment shader for example:
https://github.com/mrdoob/three.js/blob/a756b4fd1565248dfebd2e7eadb5cfc60ab986e8/examples/webgl_materials_nodes.html#L2192-L2196
Can be something like:
```javascript
Nodes.NodeLib.addKeyword( 'myCustomUv', function ( builder ) { 
  
 	if ( builder.isShader("vertex") ) return new Nodes.ReflectNode(); 
 	else return /*other node*/;
  
 } ); 
```

Reply:
👍